driver/bpf/filler_helpers.h

```diff
@@ -789,12 +799,43 @@ static __always_inline long bpf_fd_to_socktuple(struct filler_data *data,
 	return size;
 }
 
+static __always_inline int __bpf_read_val_into(struct filler_data *data,
+					       unsigned long curoff_bounded,
+					       unsigned long val,
+					       volatile u16 read_size,
+					       enum read_memory mem)
+{
+	int rc;
+	int read_size_bound;
+
+#ifdef BPF_FORBIDS_ZERO_ACCESS
+	if (read_size == 0)
+		return -1;
```

Comment:
Good point. Looking at the commit from @Molter73 ... this goes back to an `if(read_size)`. However, returning directly instead of copying zero bytes sounds good to me.

Reply:
Yeah this check is exactly what we were doing here https://github.com/falcosecurity/libs/pull/809/files#diff-c8f0685e0b19ed879c6a46c5570ae4e960346c349f9e477f2127542d6daa6c85L873

Reply:
Would that mean that we return an error when we try to read 0 bytes in a system that allows it? That doesn't sound quite right to me.

Reply:
let's say we return an `rc=-1` from that function but after this call we have
```c
if (rc)
{
	len=0;
	break;
}
```

Reply:
it is just a way to interrupt the flow immediately

Reply:
For `BPF_FORBIDS_ZERO_ACCESS`, it is considered as an error. The question actually is whether better to return zero in general and do this (in a follow) on the other places as well? So finally, having unified for the non-~ and `BPF_FORBIDS_ZERO_ACCESS` cases?

Reply:
yeah it would amazing to unify all the 2 cases in my opinion